src/__internal__/popover/popover.test.tsx

```diff
@@ -0,0 +1,159 @@
+    </Popover>
+  );
+
+  expect(screen.getByText("I float!")).not.toBeVisible();
+});
+
+test("renders popup in the document body to avoid any ancestors affecting its layout and styling", () => {
+  render(
+    <main>
+      <PopoverWithButton />
+    </main>
+  );
+
+  expect(document.body).toHaveTextContent(/I float!/);
+  expect(screen.getByRole("main")).not.toHaveTextContent(/I float!/);
+});
+
+test("renders popup within the component's ancestor as normal when disablePortal prop is true", () => {
+  render(
+    <main>
+      <PopoverWithButton disablePortal />
+    </main>
+  );
+
+  expect(screen.getByRole("main")).toHaveTextContent(/I float!/);
+});
+
+test("when mounted, renders popup in the reference's nearest dialog ancestor if it has one", async () => {
+  const DialogWithPopover = () => {
+    const [isOpen, setIsOpen] = React.useState(false);
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+
+    return (
+      <dialog open>
+        <button
+          ref={buttonRef}
+          type="button"
+          onClick={() => setIsOpen((prev) => !prev)}
+        >
+          Toggle popover
+        </button>
+        {isOpen && (
+          <Popover reference={buttonRef}>
+            <div>I float!</div>
+          </Popover>
+        )}
+      </dialog>
+    );
+  };
+  const user = userEvent.setup();
+  render(<DialogWithPopover />);
+
+  // mount the popover
+  await user.click(screen.getByRole("button", { name: /Toggle popover/ }));
+
+  expect(screen.getByRole("dialog")).toHaveTextContent(/I float!/);
+});
+
+test("applies popup styling to the element specified via the childRefOverride prop", async () => {
+  const UsingChildRefOverride = () => {
+    const buttonRef = React.useRef<HTMLButtonElement>(null);
+    const floatingRef = React.useRef<HTMLDivElement>(null);
+
+    return (
+      <main>
+        <button ref={buttonRef} type="button">
+          Toggle popup
+        </button>
+        <Popover reference={buttonRef} childRefOverride={floatingRef}>
+          <article>
+            <div ref={floatingRef}>I float!</div>
+          </article>
+        </Popover>
+      </main>
+    );
+  };
+  const user = userEvent.setup();
+  render(<UsingChildRefOverride />);
+
+  await user.click(screen.getByRole("button", { name: /Toggle popup/ }));
+
+  expect(screen.getByRole("article")).not.toHaveAttribute(
+    "data-floating-placement"
+  );
+  expect(screen.getByText("I float!")).toHaveAttribute(
+    "data-floating-placement"
+  );
+});
+
+test("renders popup within a transparent backdrop to prevent scrolling outside the popup when disableBackgroundUI is true", () => {
+  render(
+    <main>
+      <PopoverWithButton disableBackgroundUI />
+      <p>Lorem ipsum dolor sit amet...</p>
+    </main>
+  );
+
+  const backdrop = screen.getByTestId("popup-backdrop");
+  expect(backdrop).toHaveStyle("background: transparent");
+  expect(backdrop).toHaveStyle(`z-index: ${baseTheme.zIndex.popover}`);
+});
```

Comment:
I recognise that this test doesn’t verify the actual scroll-blocking behaviour. Ideally, that would be the preferred approach. However, I encountered difficulties implementing a solution that works within JSDOM. If anyone has suggestions or alternative approaches, I’d appreciate the input.

Reply:
I think what you've done is fine, as I can't think of a better way either. Maybe a Playwright test would be better - I don't think we need to test these styles in Jest for coverage, although I could be wrong - but I see there isn't currently a Playwright test file for this component, and it's probably not worth introducing one just for this.

I do have some small suggestions though to improve this test (imo) as it stands:
1) I would also test for the `position: fixed` style, with the positioning style properties (`top` etc) all set to 0, as that is what keeps it fixed to the entire viewport, which in combination with the high z-index is what prevents the background from being scrollable. (At least I think so - I haven't experimente.)
2) `toHaveStyle` [can also take an object as an argument](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohavestyle) with all the styles that you want to test. This feels less awkward to me than testing strings of CSS, and also allows all the styles to be tested in a single assertion rather than having to repeat the assertion for each style being tested.